src/main/java/ee/carlrobert/codegpt/action/ActionsUtil.java

```diff
@@ -43,7 +51,23 @@ public static void refreshActions(Map<String, String> tableData) {
       tableData.forEach((action, prompt) -> group.add(new BaseAction(action) {
         @Override
         protected void actionPerformed(Project project, Editor editor, String selectedText) {
-          sendMessage(project, prompt.replace("{{selectedCode}}", format("\n\n%s", selectedText)));
+          Pattern pattern = Pattern.compile("def\\s+(self\\.)?(\\w+)");
+
+          // Create a matcher object for the given string
+          Matcher matcher = pattern.matcher(selectedText);
+          String functionName = "";
+
+          // Check if the pattern matches and extract the function name
+          if (matcher.find()) {
+             functionName = matcher.group(2);
+            System.out.println("Function name: " + functionName);
+          } else {
+            System.out.println("no function name");
+          }
```

Comment:
**Suggestion:** Avoid using System.out.println for logging. It's better to use a logging framework like log4j or slf4j. This will give you more flexibility to configure what is logged, and it can also improve performance.
```suggestion
          import org.slf4j.Logger;
          import org.slf4j.LoggerFactory;

          ...

          private static final Logger logger = LoggerFactory.getLogger(ActionsUtil.class);

          ...

          if (matcher.find()) {
             functionName = matcher.group(2);
            logger.info("Function name: {}", functionName);
          } else {
            logger.info("no function name");
          }
```